### PR TITLE
RHICOMPL-2763: implemented the compliance aggregator

### PR DESCRIPTION
**Pull request by skateman**
From the test output:
```json
{
  "compliance": {
    "report-upload-failed": [
      {
        "policy_condition": "not-used-condition",
        "policy_description": "not-used-desc",
        "policy_id": "policy1",
        "policy_name": "not-tested-name"
      },
      {
        "policy_condition": "not-used-condition",
        "policy_description": "not-used-desc",
        "policy_id": "policy2",
        "policy_name": "not-tested-name"
      }
    ],
    "compliance-below-threshold": [
      {
        "policy_condition": "not-used-condition",
        "policy_description": "not-used-desc",
        "policy_id": "policy3",
        "policy_name": "not-tested-name"
      },
      {
        "policy_condition": "not-used-condition",
        "policy_description": "not-used-desc",
        "policy_id": "policy4",
        "policy_name": "not-tested-name"
      }
    ]
  },
  "start_time": null,
  "end_time": null
}

```

### CODE REVIEW

File: engine/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/ComplianceEmailAggregator.java [hunk 1]
```diff
@@ -1,16 +1,55 @@
...
+
+import java.util.Arrays;
+import java.util.List;
 
 public class ComplianceEmailAggregator extends AbstractEmailPayloadAggregator {
 
+    private static final String EVENT_TYPE = "event_type";
+    private static final String REPORT_UPLOAD_FAILED = "report-upload-failed";
+    private static final String SYSTEM_NON_COMPLIANT = "compliance-below-threshold";
+    // private static final String SYSTEM_NOT_REPORTING = "system-not-reporting";
```
Comment (bennetelli) [inline engine/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/ComplianceEmailAggregator.java:15]: could be removed

File: engine/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/ComplianceEmailAggregator.java [hunk 2]
```diff
@@ -1,16 +1,55 @@
...
+
+    private static final List<String> EVENT_TYPES = Arrays.asList(REPORT_UPLOAD_FAILED, SYSTEM_NON_COMPLIANT);
+    // private static final List<String> EVENT_TYPES = Arrays.asList(REPORT_UPLOAD_FAILED, SYSTEM_NON_COMPLIANT, SYSTEM_NOT_REPORTING);
```
Comment (bennetelli) [inline engine/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/ComplianceEmailAggregator.java:18]: this as well

**Comment by skateman** [comment]
Regarding the commented lines, we're planning to implement a new notification type soon and would be nice to reduce the amount of thinking for us by keeping the new type commented in the aggregator if that's okay.

File: engine/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/ComplianceEmailAggregator.java [hunk 3]
```diff
@@ -1,16 +1,55 @@
...
+        if (!EVENT_TYPES.contains(eventType)) {
+            return;
+        }
 
+        notificationJson.getJsonArray(EVENTS_KEY).stream().forEach(eventObject -> {
+            JsonObject event = (JsonObject) eventObject;
+            JsonObject payload = event.getJsonObject(PAYLOAD_KEY);
+
+            JsonArray collection = compliance.getJsonArray(eventType);
+            collection.add(payload);
```
Comment (josejulio) [inline engine/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/ComplianceEmailAggregator.java:52]: Would it be possible to have the same event twice in here?
If so, would you display it every time it appears or do you have plans to group them?

i.e.
```
// same policy for host is triggered twice in the same day?
aggregator.aggregate(ComplianceTestHelpers.createEmailAggregation("tenant", "rhel", "compliance", "compliance-below-threshold", "policy3", "host3"));
aggregator.aggregate(ComplianceTestHelpers.createEmailAggregation("tenant", "rhel", "compliance", "compliance-below-threshold", "policy3", "host3"));
```

Comment (skateman) [reply engine/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/ComplianceEmailAggregator.java:52]: All our events are unique, so we don't want to group them.

**Review by josejulio — APPROVED** [review]

**Pull request merged**